flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -211,46 +223,83 @@ protected void releaseInternal() {
 		}
 	}
 
-	private BufferBuilder getSubpartitionBufferBuilder(int targetSubpartition) throws IOException {
-		final BufferBuilder bufferBuilder = subpartitionBufferBuilders[targetSubpartition];
-		if (bufferBuilder != null) {
-			return bufferBuilder;
+	private BufferBuilder appendDataToSubpartitionForNewRecord(
+			ByteBuffer record, int targetSubpartition) throws IOException {
+		BufferBuilder buffer = subpartitionBufferBuilders[targetSubpartition];
+
+		if (buffer == null) {
+			buffer = requestNewSubpartitionBufferBuilder(targetSubpartition);
+			subpartitions[targetSubpartition].add(buffer.createBufferConsumerFromBeginning(), 0);
```

Comment:
Yes, it could be. In fact, in all existing cases, the consumer is created from the beginning. I was trying to remove `createBufferConsumer()` to use `createBufferConsumerFromBeginning()` in all places. 

In fact, I am not happy with `createBufferConsumerFromBeginning()` at all. I think I probably have a way to get rid of createBufferConsumerFromBeginning(), with a bit more complicated clean-up logic.

Reply:
I've actually tried a bit to remove `createBufferConsumerFromBeginning`, which can work, but the clean-up logic is not as beautiful as it is now: `skip() + build()`. Reason: skip can be unsuccessful because a partial record may not been written or visible yet before skip(), but can be visible after skip().

So I will keep what we have right now.

To your question:
`could it be createBufferConsumer()?`

The answer is yes to create a new record, but I think the meaning is more clear to use `createBufferConsumerFromBeginning`. What do you think?

I can definitely change it to `createBufferConsumer()`

Reply:
> The answer is yes to create a new record, but I think the meaning is more clear to use createBufferConsumerFromBeginning. What do you think?

👍 Ok, let's keep it as it is.